pinot-common/src/main/java/com/linkedin/pinot/common/utils/EqualityUtils.java

```diff
@@ -136,6 +109,6 @@ public static int hashCodeOf(int previousHashCode, int value) {
   }
 
   public static int hashCodeOf(int previousHashCode, long value) {
-    return 31 * previousHashCode + (int) (value ^ (value >>> 32));
+    return 31 * previousHashCode + (int) ((value & 0xFFFFFFFFL) ^ (value >>> 32));
```

Comment:
I think the purpose of the second part is to do XOR on high 32 bits and low 32 bits for long type. But the conversion from long to int will set the sign bit for int, and increase the chance of collision.